iwa-console/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,270 @@
+            this.PublicClientApplication = this.PublicClientApplicationBuilder
+                    .WithBrokerPreview(true)
+                    .WithParentActivityOrWindow(() => { return handle.Value; })                                     // Specify Window handle - (required for WAM).
+                    .Build();
+
+            this.IsBrokerInitialized = true;
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Attaches the token cache to the Public Client app.
+        /// </summary>
+        /// <returns>IAccount list of already signed-in users (if available)</returns>
+        private async Task<IEnumerable<IAccount>> AttachTokenCache()
+        {
+            // Cache configuration and hook-up to public application. Refer to https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/wiki/Cross-platform-Token-Cache#configuring-the-token-cache
+            var storageProperties = new StorageCreationPropertiesBuilder(AzureADConfig.CacheFileName, AzureADConfig.CacheDir).Build();
+            var msalcachehelper = await MsalCacheHelper.CreateAsync(storageProperties);
+            msalcachehelper.RegisterCache(this.PublicClientApplication.UserTokenCache);
+
+            // If the cache file is being reused, we'd find some already-signed-in accounts
+            return await this.PublicClientApplication.GetAccountsAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signs in the user and obtains an Access token for a provided set of scopes
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns> Access Token</returns>
+        public async Task<string> SignInUserAndAcquireAccessToken(string[] scopes)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+
+            try
+            {
+                // 1. Try to sign-in the previously signed-in account
+                if (existingUser != null)
+                {
+                    this.AuthResult = await this.PublicClientApplication.AcquireTokenSilent(scopes, existingUser)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    if (this.IsBrokerInitialized)
+                    {
+                        Console.WriteLine("No accounts found in the cache. Trying Window's default account.");
+
+                        this.AuthResult = await this.PublicClientApplication
+                            .AcquireTokenSilent(scopes, Microsoft.Identity.Client.PublicClientApplication.OperatingSystemAccount)
+                            .ExecuteAsync()
+                            .ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        this.AuthResult = await SignInUserInteractivelyAsync(scopes);
```

Comment:
Won't be needed in broker only, since `AcquireTokenSilent` will throw `MsalUiRequiredException `